src/Datadog.Trace.ClrProfiler.Managed/Emit/MethodBuilder.cs

```diff
@@ -107,20 +108,27 @@ public MethodBuilder<TDelegate> WithReturnType(Type returnType)
 
         public TDelegate Build()
         {
+            var parameterTypesForCache = _explicitParameterTypes;
+
+            if (parameterTypesForCache == null)
+            {
+                parameterTypesForCache = Interception.ParamsToTypes(_parameters);
+            }
```

Comment:
Found failures from the cache key being missed when we didn't specify explicit types.